R/utilities-units.R

```diff
@@ -48,9 +48,10 @@ toUnit <- function(quantity, values, targetUnit) {
   validateIsOfType(quantity, Quantity)
   validateIsNumeric(values)
   validateHasUnit(quantity, targetUnit)
+  targetUnitIndex <-  as.integer(quantity$unitIndexOf(targetUnit))
   values <- c(values)
   sapply(values, function(value) {
-    rClr::clrCallStatic(WITH_DIMENSION_EXTENSION, "ConvertToUnit", quantity$ref, value, targetUnit)
+    rClr::clrCallStatic(R_WITH_DIMENSION_EXTENSION, "ConvertToUnit", quantity$ref, value,targetUnitIndex)

```

Comment:
I am using the index here so that the string conversion does not happen in rClr but in .NET
Quite crazy but this changes is very much localized . We also always validate that the unit as string exists with `validateHasUnit(quantity, unit)` so we should be good to go